syntax = "proto3";

package web_service;

message Empty {}

message PositionRequest {
  float pos_x_m = 1;
  float pos_y_m = 2;
}

message OrientationRequest {
  float orientation_rad = 1;
}

message RobotDataRequest {
  float pos_x_m = 1;
  float pos_y_m = 2;
  float orientation_rad = 3;
  float battery_v = 4;
  bool is_grabber_open = 5; 
}

message SpeedRequest {
  int32 motor1 = 1;
  int32 motor2 = 2;
}

service Position{
  rpc setSpeedRequest(SpeedRequest) returns (Empty) {}
  rpc setAbsolutePositionRequest(RobotDataRequest) returns (Empty) {}
  rpc setTargetPositionRequest(PositionRequest) returns (Empty) {}
  rpc setTargetOrientationRequest(OrientationRequest) returns (Empty) {}
  rpc registerRobotDataObserver(Empty) returns (stream RobotDataRequest) {}
}
